src/main/java/hu/wtomee/objects/DFA.java

```diff
@@ -94,16 +107,38 @@ public DFA resetState() {
         return this;
     }
 
-    public DFA input(String input) {
-        if (!alphabet.contains(input))
-            throw new InvalidParameterException("\"" + input + "\" is not valid");
+    public void getAccessiblePart() {
+        List<String> accessibleStates = new ArrayList<>();
+        accessibleStates.add(initialState);
+        currentState = initialState;
+        for (int i = 0; i < transitions.size(); i++) {
+            currentState = transitions.stream()
+                    .filter(transition -> transition.getFromState().equals(currentState))
+                    .findFirst()
+                    .get().getNextState();
+            if(!accessibleStates.contains(currentState))
+                accessibleStates.add(currentState);
+        }
+        states = accessibleStates;
 
+        System.out.println("Accessible states: " + states);
 
-        if(transitions.stream().noneMatch(transition -> transition.getSymbol().equals(input) && transition.getFromState().equals(currentState)))
-            throw new InvalidParameterException("No transition found for input: " + input);
+        transitions = transitions.stream()
+                .filter(transition -> accessibleStates.contains(transition.getNextState())).toList();
```

Comment:
Ezeket a stream-es magic-eket nem ismerem annyira, de nem azok a transition-ok maradnak bent, amik accessible state-ekbe mutatnak, hanem amik ilyenbol mutatnak.

`>s --a--> p <--a-- q`

Itt ugye `{s,p}` az accessible, de a Te filtered alapjan `(q,a,p)` is bentmarad.